Added Evaluation Calendar Components

saikrishnamantri opened the pull request:
Components:
- CalendarView.tsx – Parent component that organizes and passes grouped data
- CalendarDayCard.tsx – Reusable component that renders evaluations for a single day

achow-conestogac commented:
Before this can be reviewed, I will need to see tests and documentation.

achow-conestogac commented on src/Components/EvaluationCalendar/CalendarDayCard.tsx:
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { Evaluation } from "./Evaluation";
```
this doesn't work.

no such thing as `./Evaluation`

achow-conestogac commented on src/Components/EvaluationCalendar/CalendarDayCard.tsx:
```diff
@@ -0,0 +1,32 @@
+        <div className="bg-white shadow rounded p-4">
+            <h2 className="text-xl font-bold mb-2">{date}</h2>
+            <ul className="space-y-2">
+                {evaluations.map((ev, idx) => (
+                    <li
+                        key={idx}
+                        className="border border-gray-200 rounded p-2 hover:bg-gray-50"
+                    >
+                        <div className="text-sm font-semibold">
+                            {ev.title} ({ev.type})
+                        </div>
+                        <div className="text-xs text-gray-600">
+                            Course: {ev.course} | Weight: {ev.weight}%
+                        </div>
+                    </li>
```
what happens if you don't have evaluation data? do i just end up with a big blank screen?

please apply good error handling.  you should do a check to see if this data is available and then proceed.  If it is not available, you should return some sort of message telling the using that there's no data (or whatever the approrpiate response may be)

saikrishnamantri replied:
Updated CalendarDayCard.tsx with proper error handling and a user-friendly message.

SreejaAlla commented:
Added Evaluation.ts - Evaluation Data Model
  
  Tests done for CalendarDayCard
    √ renders the correct date heading.                                                                                                                       
    √ displays all evaluations for the day.                                                                                                                    
    √ displays fallback message when no evaluations exist.

saikrishnamantri commented:
Fix: date shifting across timezones to ensure consistent date-formatting

- Replaced .toDateString() with Intl.DateTimeFormat to handle timezone discrepancies.
- Updated CalendarView.test.tsx to use the same date formatter as the component.
- Fallback "No evaluations" message for better UX on empty state.
- Improves test reliability across local and CI environments.

Tests:
- √ renders a CalendarDayCard for each unique date.
- √ renders evaluations under the correct date.
- √ shows a message when no evaluations are provided.

achow-conestogac commented:
Your file should be arranged something like this.  The following is a snippet of a comment i pulled from another PR so the file names are not right.

```
| src
| -- Components
|---- ConfirmationMessage
|------ConfirmationMessage.tsx
|------ConfirmationMessage.test.tsx
```

This should illustrate the purpose though. The module (folder) that your code is in really tells the user (in this case, other developers) what the point of that folder is.  And as another developer, I really shouldn't need to dig into the folder to know what it is for.

The `Component` directory is used for generic, reusable components.  And in this case, it is `CalendarDayCard.tsx`.  

The `CalendarView` seems out of place.  This is an _implementation_ which consumes `CalendarDayCard.tsx`.  this really should be done in the "page" where your logic comes together.  If you don't know how the folder structure is going to be at this moment, I highly suggest you simply submit the `CalendarDayCar.tsx` only (associated tests, of course) in this PR.

saikrishnamantri commented:
feat: add CalendarDayCard component with unit tests

- Removed other files that doesn't required
- Implemented CalendarDayCard to display evaluations for a single day
- Included fallback message for empty evaluation lists
- Added unit tests to verify rendering, data display, and empty state handling

saikrishnamantri commented on src/Components/EvaluationCalendar/CalendarView.tsx:
- Added index.ts file and documentation for CalendarDayCard. 
- Deleted the empty file and folder.
